### PR TITLE
Add Singularity testing to Travis CI

### PR DESCRIPTION
**Pull request by MaxUlysse**
- new `singularity.config` file
- new `singularity` profile
- add `install.sh` script to install Nextflow and/or Singularity
- add `env` to travis testing (might be a good idea to control with which Nextflow/Singularity version we test).
- now we're testing NGI-RNAseq with Nextflow/Docker and with Nextflow/Singularity

### CODE REVIEW

**Comment by MaxUlysse** [comment]
Can't use `wf_container` as wanted in `singularity.config`, so defining the container in `singularity.config`

**Comment by Hammarn** [comment]
I'm happy with this. The only small thing is that I'm not sure we want to specify the Nextflow version, I can see an advantage in it updating automatically. That's trivial to revert back to later if that's the case.
Good work @MaxUlysse 👍

**Comment by MaxUlysse** [comment]
I think it's safer to specify the Nexflow version, that way, you're sure that no Nextflow update will break the pipeline.
And actually, you can probably still add another tests with the latest Nextflow version too ;-)

File: conf/singularity.config [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * NOTE: Not suitable for production use, assumes
+ * compute limits of only 16GB memory and 1 CPU core.
+ */
+
+singularity {
+  enabled = true
+}
+
+process {
+  container = { "docker://scilifelab/ngi-rnaseq:${workflow.revision ? workflow.revision : 'latest'}" }
```
Comment (ewels) [inline conf/singularity.config]: Why couldn't you use `wf_container` here?

Comment (ewels) [reply conf/singularity.config]: Should be no different to the [`uppmax` profile](https://github.com/SciLifeLab/NGI-RNAseq/blob/master/conf/uppmax.config#L19), no?

Comment (MaxUlysse) [reply conf/singularity.config]: No idea at all, when I tried "docker://$wf_container" all I got was: "docker://" without anything after...
Same with "docker://${wf_container}"...
Does that really work in the uppmax profile?

Comment (ewels) [reply conf/singularity.config]: We changed it recently, so it's possible that the change broke it. But it should do, yes. The docker profile used for automated testing uses it too: https://github.com/SciLifeLab/NGI-RNAseq/blob/master/conf/docker.config#L19

Comment (MaxUlysse) [reply conf/singularity.config]: I know that in the docker profile it works.
But it seems to me, that you can use the variable as it is, but you can't use it.
I don't really understand.
I'll try to make a reproducible minimal example to make an issue for Paolo

Comment (ewels) [reply conf/singularity.config]: Should be v. easy to run the uppmax test run script to check whether the uppmax config file actually works as expected.

Comment (Hammarn) [reply conf/singularity.config]: It is indeed very simple and it does currently not work:

Comment (ewels) [reply conf/singularity.config]: Ah, I bet we need to defer the execution of this line as well:

```groovy
container = { "docker://$wf_container" }
```

Comment (MaxUlysse) [reply conf/singularity.config]: Totally, but how do you intend to do that?

Comment (ewels) [reply conf/singularity.config]: With the snippet in my comment above I think.. Just use squiggly brackets around the entire expression.

Comment (MaxUlysse) [reply conf/singularity.config]: Oh I see...

Comment (Hammarn) [reply conf/singularity.config]: @ewels I tried what you suggested with no luck. Any other ideas?

Comment (MaxUlysse) [reply conf/singularity.config]: I'm not sure it's possible at all to do that right now with Nextflow.
I'll make an issue about it...

**Comment by MaxUlysse** [comment]
@Hammarn 
Can you try that?
cf https://github.com/nextflow-io/nextflow/issues/602#issuecomment-363492875

**Pull request merged**